packages/web/components/Quest/QuestFilter.tsx

```diff
@@ -34,89 +39,212 @@ export const QuestFilter: React.FC<Props> = ({
+        />
+        <MetaFilterSelectSearch
+          title={`Order: ${order.label}`}
+          tagLabel=""
+          styles={metaFilterSelectStyles}
+          hasValue={false}
+          value={order}
+          onChange={(value) => {
+            const values = value as ValueType[];
+            const o = values[values.length - 1];
+            if (o) {
+              setOrder(o);
+              setQueryVariable('order', o.value);
+            }
+          }}
+          options={orderOptions}
+          disableEmpty
+        />
+        <MetaFilterSelectSearch
+          title={`Status: ${status.label}`}
+          tagLabel=""
+          styles={metaFilterSelectStyles}
+          hasValue={false}
+          value={status}
+          onChange={(value) => {
+            const values = value as ValueType[];
+            const s = values[values.length - 1];
+            if (s) {
+              setStatus(s);
+              setQueryVariable('status', s.value);
+            }
+          }}
+          options={statusOptions}
+          disableEmpty
+        />
+        {aggregates.guilds.length && (
+          <MetaFilterSelectSearch
+            title={`Guild: ${guild.label}`}
+            tagLabel=""
+            styles={metaFilterSelectStyles}
+            hasValue={false}
+            value={guild}
+            onChange={(value) => {
+              const values = value as ValueType[];
+              const g = values[values.length - 1];
+              if (g) {
+                setGuild(g);
+                setQueryVariable('guild_id', g.value);
               }
-            >
-              <option value={10}>10</option>
-              <option value={20}>20</option>
-              <option value={50}>50</option>
-            </MetaSelect>
-          </WrapItem>
-          <WrapItem>
-            <MetaSelect
-              value={queryVariables.order as string}
-              onChange={(e) => setQueryVariable('order', e.target.value)}
-            >
-              <option value={Order_By.Desc}>Newest</option>
-              <option value={Order_By.Asc}>Oldest</option>
-            </MetaSelect>
-          </WrapItem>
-          <WrapItem>
-            <MetaSelect
-              value={queryVariables.status as string}
-              onChange={(e) => setQueryVariable('status', e.target.value)}
-            >
-              <option value={QuestStatus_Enum.Open}>Open</option>
-              <option value={QuestStatus_Enum.Closed}>Closed</option>
-            </MetaSelect>
-          </WrapItem>
+            }}
+            options={guildOptions}
+            disableEmpty
+          />
+        )}
+        {roleChoices.length && (
+          <MetaFilterSelectSearch
+            title="Roles"
+            tagLabel={roles.length > 0 ? roles.length.toString() : ''}
+            styles={metaFilterSelectStyles}
+            hasValue={false}
+            value={roles}
+            onChange={(value) => {
+              const values = value as ValueType[];
```

Comment:
Does this work for setting the variable name and type? If so, I would use this format on all the filters.

```suggestion
            onChange={(values: Array<ValueType>) => {
```

Reply:
Fixing this would require digging into the type structure and arguments of `MetaFilterSelectSearch`. It is known that it takes an array of values, so the argument should be called `values` instead of `value`. The type passed into `onChange` needs some work as well.

It's not central to what you're doing at all & ought to be left for another day.